File: backend/internal/db/models/recipe.go
package models

type Recipe struct {
	ID uint64 `gorm:"autoIncrement"`
	Name string `gorm:"not null"`
	UniqueName string `gorm:"unique;not null"`
	Description string
	OvenTemp int
	EstimatedTime int
}

func (_ Recipe) StructName() string {
	return "Recipe"
}

File: nextjs-frontend/src/api/Recipe.ts
import { Step } from "./Step";
import { Ingredient } from "./Ingredient";
import { Image } from "./Image";
import { Author } from "./Author";
import { Tag } from "./Tag";

export interface Recipe {
  id: string;
  name: string;
  uniqueName: string;
  description: string;
  ovenTemperature: string;
  estimatedTime: string;
  steps: Step[];
  ingredients: Ingredient[];
  images: Image[];
  author: Author;
  tags: Tag[];
}
